firedrake/solving_utils.py

```diff
@@ -21,6 +23,41 @@ def _make_reasons(reasons):
 SNESReasons = _make_reasons(PETSc.SNES.ConvergedReason())
 
 
+if get_config()["options"]["petsc_int_type"] == "int32":
+    DEFAULT_KSP_PARAMETERS = {"mat_type": "aij",
+                              "ksp_type": "preonly",
+                              "pc_type": "lu",
+                              "pc_factor_mat_solver_type": "mumps"}
+else:
+    # FIXME: which factorisation package to use in int64?
+    DEFAULT_KSP_PARAMETERS = {}
```

Comment:
The MUMPS manual suggests it can be built in fully 64-bit mode, and in "partial" 64-bit mode (where only things that might be big are 64-bit integers). Does PETSc not support it?

Reply:
PETSc buildsystem says:
```python
    # Mumps does NOT work with 64 bit integers without a huge number of hacks we ain't making
    self.precisions       = ['single','double']
    self.requires32bitint = 1;  # 1 means that the package will not work with 64 bit integers
```

Reply:
Fair enough. I guess the next best choice would be `superlu_dist`, but that would require changing the default PETSc configure options too.

Reply:
#1474 changes petsc defaults and installs superlu_dist by default, so if that lands, I will do that here.